src/main/java/com/google/sps/data/DatastoreClient.java

```diff
@@ -0,0 +1,136 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import com.google.sps.data.Attendee;
+import com.google.sps.data.AttendeeInterface;
+import com.google.sps.data.Session;
+import com.google.sps.data.SessionInterface;
+import com.google.sps.data.Instance;
+import com.google.sps.data.InstanceInterface;
+
+/** Class that reads from and writes to Datastore. */
+public class DatastoreClient implements DatastoreClientInterface {
+  /** 
+   * Adds an attendeeInterface object to the Datastore. If object already
+   * exists, the old one is replaced. 
+   */
```

Comment:
You dont need to repeat the function comment in the implementation. Just have it in the interface